doc/usage/configuration.rst

```diff
@@ -2522,6 +2522,28 @@ Options for the linkcheck builder
 
    .. versionadded:: 1.1
 
+.. confval:: linkcheck_allowed_redirects
+
+   A dictionary that maps a pattern of the source URI to a pattern of the canonical
+   URI. The linkcheck builder treats the redirected link as "working" when:
+
+    - the link in the document matches the source URI pattern, and
+    - the redirect location matches the canonical URI pattern.
+
+   Example:
+
+   .. code-block:: python
+
+      linkcheck_working_redirects = {
+          # All HTTP redirections from the source URI to the canonical URI will be treated as "working".
+          r'https://sphinx-doc\.org/.*': r'https://sphinx-doc\.org/en/master/.*'
```

Comment:
I think the example should use patterns for both source and canonical URIs. So I think the current one is enough to describe the feature.

Reply:
But this is not just allowing redirects to the canonical version of a URL, it's allowing any redirects.

You may as well use `r'https://sphinx-doc\.org/.*': r'https://sphinx-doc\.org/.*'` because there isn't really any validation of the destination.

Reply:
What do you mean? The current rule verifies the pages under `https://sphinx-doc.org/` to the URL under `https://sphinx-doc.org/en/master/`.  It does not mean "any redirects".

Reply:
It's allowing practically any redirect, so not what I would consider a good example.

It's not far off allowing `https://www\.example\.com/.*` to `https://www\.example\.com/.*` which then causes linkcheck to ignore removed pages that just redirect to the home page.

Reply:
If `https://sphinx-doc.org/foo/bar` redirects to `https://sphinx-doc.org/other/path`, it's not "allowed". Is this allowing any redirect? I'm confused.